packages/fuels-contract/src/contract.rs

```diff
@@ -384,11 +384,26 @@ impl Contract {
 
         match provider.client.submit(&tx).await {
             Ok(_) => Ok(contract_id),
-            Err(e) => Err(Error::TransactionError(e.to_string())),
+            Err(e) => Err(Error::TransactionError(format!(
+                "{} {}",
+                e.to_string(),
+                "-> maybe use load_sway_contract_with_salt",
```

Comment:
I think this message will, by and large, be very misleading. 

The overwhelming majority of error cases aren't going to be related to the salt. For instance, it could be due to bad network conditions or poorly formatted transactions. So, in most cases, people will be led to believe that using `load_sway_contract_with_salt()` could make a difference, when in most cases it won't, making it a very frustrating experience. 

I'd say remove the suggestion altogether or parse the error, and, if it has something to do with ID collision, suggest using the load with salt method.

Reply:
That's why I made issue #249 . In this case, i do not get the error type but the string type that is sent from the FuelClient.query function. I could parse that string and look within it for something that would indicate ID collision, but is that a good solution? Maybe it would be better if the fuel-core in the FuelClient.query function sends enum insted of a string, that would indicate that a certain problem has occurred (from GraphQlError to fuel-core.errors enum) .
Like you said, it would be best to delete suggestion "-> maybe use load_sway_contract_with_salt" until it is decided how we will proceed.